### PR TITLE
Jenkins Swarm and xenial slaves

### PR DESCRIPTION
**Pull request by fgimenez**
Added swarm slave container and k8s ReplicationController, including the required changes to make it work with the current master. The current configuration deploys 4 xenial slave containers with 2 executors each.

These changes are backwards compatible, the cloud and local imperative deployments have been updated to work with the new config.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Snappy Jenkins Docker container
+# Snappy Jenkins CI
 
-This repo helps setting up the CI environment used for executing Snappy integration tests [1]. It contains a Dockerfile which defines the configuration of a Jenkins container and several shell scripts for bringing it to live both in a cloud provider (only OpenStack supported at the moment) and locally.
+This repo helps setting up the CI environment used for executing Snappy integration tests [1]. It contains Dockerfiles which defines the required containers (Jenkins master and slaves and a Nginx proxy) and several shell scripts for bringing them to live both in a cloud provider (only OpenStack supported at the moment) and locally.
```
Comment (elopio) [inline README.md]: s/defines/define

File: README.md [hunk 2]
```diff
@@ -38,15 +38,15 @@ being, as in the local case, `<cloud_credentials_path>` the location of an OpeSt
...
 
-* Jenkins instance: the same as in the local provision case.
+* Jenkins master instance: the same as in the local provision case.
 
 It also setups a security group that allows access to port 8081 from everywhere and ports 8080 and 22 from a local range (by default 10.0.0.0/8). All this configuration is done this way in order to facilitate a secure connection from the GitHub webhook, as detailed in the next section.
 
 ## GitHub integration
 
-The Jenkins container has installed the GitHub Pull Request Builder Plugin [3] to allow the triggering of jobs in response to events from GitHub. The default configuration in the container leaves almost all setup, there's only a few things left to do:
+The Jenkins master container has the GitHub Pull Request Builder Plugin installed [3] to allow the triggering of jobs in response to events from GitHub. The default configuration in the container leaves almost all setup, there's only a few things left to do:
```
Comment (elopio) [inline README.md]: there ARE only a few things...

File: bin/common.sh [hunk 3]
```diff
@@ -3,11 +3,15 @@
 NAME=jenkins-master-service
 PROXY_NAME=snappy-proxy
 JENKINS_CONTAINER_NAME=fgimenez/snappy-jenkins
-JENKINS_CONTAINER_INIT_COMMAND="sudo docker run -p 8080:8080 -d -v $JENKINS_HOME:/var/jenkins_home --privileged=true --restart always --name $NAME -t $JENKINS_CONTAINER_NAME"
+JENKINS_CONTAINER_INIT_COMMAND="sudo docker run -p 8080:8080 -d -v $JENKINS_HOME:/var/jenkins_home --restart always --name $NAME -t $JENKINS_CONTAINER_NAME"
+JENKINS_MASTER_CONTAINER_DIR="./containers/jenkins-master"
 JENKINS_MASTER_CONTAINER_DIR="./containers/jenkins-master"
```
Comment (elopio) [inline bin/common.sh]: here you have the same var twice.

File: bin/common.sh [hunk 4]
```diff
@@ -3,11 +3,15 @@
...
 
 PROXY_CONTAINER_NAME="fgimenez/snappy-jenkins-proxy"
-PROXY_CONTAINER_INIT_COMMAND="sudo docker run -d -p 8081:80 --link $NAME:$NAME --restart always -v /var/run/docker.sock:/tmp/docker.sock:ro --name $PROXY_NAME $PROXY_CONTAINER_NAME"
+PROXY_CONTAINER_INIT_COMMAND="sudo docker run -d -p 8081:80 --link $NAME:$NAME --restart always --name $PROXY_NAME $PROXY_CONTAINER_NAME"
 PROXY_CONTAINER_DIR="./containers/jenkins-proxy"
 
-JENKINS_MASTER_CONTAINER_DIR="./containers/jenkins-master"
+SLAVE_BASE_NAME=jenkins-slave
+SLAVE_NAME=$SLAVE_BASE_NAME
```
Comment (elopio) [inline bin/common.sh]: where is this one used? I can't find it.

**Comment by elopio** [comment]
Other than typos, I can't see anything obviously wrong. You already deployed it, so :+1:

**Comment by fgimenez** [comment]
Good catches, already fixed thanks!

**Pull request merged**